LayoutTests/http/tests/paymentrequest/paymentrequest-merchantCategoryCode.https.html

```diff
@@ -0,0 +1,73 @@
+<script src="/js-test-resources/ui-helper.js"></script>
+<script src="/resources/payment-request.js"></script>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<body>
+<script>
+    setup({ explicit_done: true, explicit_timeout: true });
+    test((test) => {
+        const method = validPaymentMethod();
+        method.data.features = ["paymentRequestMerchantCategoryCode"];
+        new PaymentRequest([method], validPaymentDetails());
+    }, "Should have a feature for `paymentRequestMerchantCategoryCode`.");
+    
+    user_activation_test(async (test) => {
+        const request = new PaymentRequest([validPaymentMethod()], validPaymentDetails());
+        
+        request.addEventListener("merchantvalidation", (event) => {
+            event.complete({ });
+        });
+        
+        const shippingPromise = new Promise(resolve => {
+            request.addEventListener("shippingaddresschange", (event) => {
+                assert_equals(internals.mockPaymentCoordinator.merchantCategoryCode, null, "check that the `merchantCategoryCode` is still `null` after an update");
+                event.updateWith({ });
+                internals.mockPaymentCoordinator.acceptPayment();
+                
+                resolve();
+            });
+        });
+        
+        const response = await request.show();
+        
+        await shippingPromise;
+        
+        assert_equals(internals.mockPaymentCoordinator.merchantCategoryCode, null, "check that the `merchantCategoryCode` is still `null` after the payment is accepted");
+        
+        await response.complete("success");
+    }, "Should not have a default value for `merchantCategoryCode`.");
+    
+    user_activation_test(async (test) => {
+        const method = validPaymentMethod();
+        method.data.merchantCategoryCode = "0739";
+        
+        const request = new PaymentRequest([method], validPaymentDetails());
+        
+        request.addEventListener("merchantvalidation", (event) => {
+            event.complete({ });
+        });
+        
+        const shippingPromise = new Promise(resolve => {
+            request.addEventListener("shippingaddresschange", (event) => {
+                assert_equals(internals.mockPaymentCoordinator.merchantCategoryCode, method.data.merchantCategoryCode, "check that the `merchantCategoryCode` still matches after an update");
+                event.updateWith({ });
+                internals.mockPaymentCoordinator.acceptPayment();
+                
+                resolve();
+            });
+        });
+        
+        const response = await request.show();
+        
+        await shippingPromise;
+        
+        assert_equals(internals.mockPaymentCoordinator.merchantCategoryCode, method.data.merchantCategoryCode, "check that the `merchantCategoryCode` still matches after the payment is accepted");
+        
+        await response.complete("success");
```

Comment:
Maybe, just: 

```suggestion
        const shippingPromise = new Promise(resolve => {
            request.addEventListener("shippingaddresschange", resolve);
        });
        
        const response = await request.show();
        
        const event = await shippingPromise;
        assert_equals(
          internals.mockPaymentCoordinator.merchantCategoryCode,
          method.data.merchantCategoryCode, 
          "check that the `merchantCategoryCode` still matches after an update"
        );
        event.updateWith({ });
        
        internals.mockPaymentCoordinator.acceptPayment();
        assert_equals(
          internals.mockPaymentCoordinator.merchantCategoryCode, 
          method.data.merchantCategoryCode,
          "check that the `merchantCategoryCode` still matches after the payment is accepted"
        );
        
        await response.complete("success");
```

Reply:
So I tried using promises, but that causes trouble with the payment request internals. Unfortunately it expects the `updateWith` call to be done synchronously and so doesn't play nice with Promises.

I think it's ok to skip using them here, especially as none of the other payment request tests use it and the consistency might help readability for someone looking through.